pure.zsh

```diff
@@ -222,6 +222,13 @@ prompt_pure_precmd() {
 		export VIRTUAL_ENV_DISABLE_PROMPT=12
 	fi
 
+    # Nix package manager integration.
+    # If this prompt is used from within 'nix shell' - present shell derivation name as
+    # shell name. If derivation name is unset - show 'nix-shell'
```

Comment:
Tab-indentation.

It's also not clear to me as a non-nix user, what "shell derivation name" is. Maybe include an example of what the prompt looks like (as text) with "shell derivation name"?

Reply:
https://stackoverflow.blog/2017/06/15/developers-use-spaces-make-money-use-tabs/ ;)

`Nix` package manager has a concept of `derivations` which basically is a compilation unit that results in a file or directory (Can be package/script/config/etc). And derivation `name` is a user-defined value that can be anything. In the case of `mkShell` derivation builder most likely you'd set `name` argument according to your project name.

Can you please tell me what kind of comment is expected here and ill more or less copy-paste it since I am really not sure what else am I supposed to type. I feel like to explain it to non-nix users I would straight up need to copy-paste few pages of documentation, it is a very different build tool/package manager, with very different ideas compared to most. Getting a hang of it [takes some time](https://raw.githubusercontent.com/gytis-ivaskevicius/high-quality-nix-content/master/memes/nix-learning-curve.png)

Reply:
My pure prompt looks like:

```
~/dev
❯
```

I'm looking for a code comment with the above just how it looks like with the Nix stuff.

Reply:
Here is an example of how it looks:
![image](https://user-images.githubusercontent.com/23264966/125101046-74b23980-e0e2-11eb-89e4-aa048ad89a14.png)

And here is ASCII copy-paste:
```
~/Projects/flake-utils-plus github-actions*
flake-utils-plus ❯ unset name

~/Projects/flake-utils-plus github-actions*
nix-shell ❯
```

Would you like me to drop this ASCII as a comment above that if statement?

Reply:
Yes